File: 백준/Gold/3079. 입국심사/입국심사.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
time_list = list(int(input()) for _ in range(n))
time_list.sort()

def get_check_num(max_time):
    check_num = 0
    for time in time_list:
        check_num += max_time // time
    
    return check_num

def select_time():
    left = 1
    right = m * min(time_list)
    answer = sys.maxsize
    while left <= right:
        mid = (left + right) // 2
        check_num = get_check_num(mid)
        if check_num < m:
            left = mid + 1
        else:
            answer = min(answer, mid)
            right = mid - 1

    return answer
        
print(select_time())

File: 백준/Gold/3079. 입국심사/입국심사.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().trim().split('\n')

const [N, M] = input.shift().split(' ').map(Number)
const time = input.map(v => BigInt(v))

const MAX_TIME = BigInt(M) * time[N - 1]
let answer = MAX_TIME
let left = 1n
let right = MAX_TIME
while (left <= right) {
  const mid = (left + right) / 2n
  const friends = review_friends(mid)

  if (M <= friends) {
    answer = mid
    right = mid - 1n
  } else {
    left = mid + 1n
  }
}
console.log(answer.toString())

function review_friends(max_time) {
  let cnt = 0n
  for (const t of time) {
    if (M < cnt) break
    cnt += max_time / t
  }

  return cnt
}